include/boost/math/tools/polynomial.hpp

```diff
@@ -354,6 +360,12 @@ class polynomial :
    }
 
    // operators:
+   polynomial& operator =(polynomial p)
```

Comment:
We have two cases:
```cpp
polynomial<double> p{1, 2, 3, 4};

// case 1:
polynomial<double> q;
q = p; // this is handled optimally in both c++03 and c++11 since q has no memory allocated to it

// case 2:
polynomial<double> r{5, 6, 7, 8, 9, 1, 2, 3, 4};
r = p; // this is bad. We could have prevented a copy by overwriting r.m_data
```

Will fix it shortly. I'm afraid the code wil turn into a mess of `#if`s./`#ifndef`s. Can you suggest a way we can prevent this from happening?